hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/server/TopicMetadataLoadingStartupHook.java

```diff
@@ -0,0 +1,208 @@
+
+public class TopicMetadataLoadingStartupHook implements ServiceAwareHook {
+
+    private static final Logger logger = LoggerFactory.getLogger(TopicMetadataLoadingStartupHook.class);
+
+    private final BrokerMessageProducer brokerMessageProducer;
+
+    private final TopicsCache topicsCache;
+
+    private final GroupRepository groupRepository;
+
+    private final TopicRepository topicRepository;
+
+    private final int retryCount;
+
+    private final long retryInterval;
+
+    private final int threadPoolSize;
+
+    @Inject
+    public TopicMetadataLoadingStartupHook(BrokerMessageProducer brokerMessageProducer,
+                                           TopicsCache topicsCache,
+                                           GroupRepository groupRepository,
+                                           TopicRepository topicRepository,
+                                           ConfigFactory config) {
+        this(brokerMessageProducer,
+                topicsCache,
+                groupRepository,
+                topicRepository,
+                config.getIntProperty(Configs.FRONTEND_STARTUP_TOPIC_METADATA_LOADING_RETRY_COUNT),
+                config.getLongProperty(Configs.FRONTEND_STARTUP_TOPIC_METADATA_LOADING_RETRY_INTERVAL),
+                config.getIntProperty(FRONTEND_STARTUP_TOPIC_METADATA_LOADING_THREAD_POOL_SIZE));
+    }
+
+    public TopicMetadataLoadingStartupHook(BrokerMessageProducer brokerMessageProducer,
+                                           TopicsCache topicsCache,
+                                           GroupRepository groupRepository,
+                                           TopicRepository topicRepository,
+                                           int retryCount,
+                                           long retryInterval,
+                                           int threadPoolSize) {
+        this.brokerMessageProducer = brokerMessageProducer;
+        this.topicsCache = topicsCache;
+        this.groupRepository = groupRepository;
+        this.topicRepository = topicRepository;
+        this.retryCount = retryCount;
+        this.retryInterval = retryInterval;
+        this.threadPoolSize = threadPoolSize;
+    }
+
+    @Override
+    public void accept(ServiceLocator serviceLocator) {
+        long start = System.currentTimeMillis();
+        logger.info("Loading topics metadata");
+        ExecutorService executorService = Executors.newFixedThreadPool(threadPoolSize);
+        try {
+            loadTopicsMetadataInExecutor(executorService);
+        } finally {
+            executorService.shutdown();
+        }
+        logger.info("Loading topics metadata done in {}ms", System.currentTimeMillis() - start);
+    }
+
+    private void loadTopicsMetadataInExecutor(ExecutorService executorService) {
+        List<TopicName> topics = getTopics();
+
+        List<LoadingResult> successes = new ArrayList<>();
```

Comment:
what is the difference between `successes` and `failures` that it is constructed in different way? :)